File: EcoZone.Domain/Models/Article.cs
using System;
using System.Collections.Generic;

namespace EcoZone.Domain.Models
{
    public class Article
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Source { get; set; }
        public string ImagePath { get; set; }
        public int Views { get; set; } = 0;
        public DateTime Date { get; set; } = DateTime.Now;
        public bool IsApproved { get; set; } = false;

        public User User { get; set; }
        public string UserId { get; set; }
        public Article PendingArticle { get; set; }
        public int? PendingArticleId { get; set; }

        public ICollection<Tag> Tags { get; set; }
        public ICollection<Comment> Comments { get; set; }
        public ICollection<Like> Likes { get; set; }
    }
}

File: EcoZone.Front/src/app/models/article/article.ts
import { User } from '../user';
import { Comment } from '../comment';
import { Tag } from '../tag';
import { Like } from '../like';

export class Article {
    private id: number;
    private title: string;
    private description: string;
    private source: string;
    private imagePath: string;
    private views: number;
    private date;
    private isApproved: boolean;
    private user: User;
    private tags: Array<Tag>;
    private comments: Array<Comment>;
    private likes: Array<Like>;

    constructor(id) {
        this.id = id;
        this.title = 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
            'Distinctio sapiente ratione officiis laboriosam, ' +
            'ipsum quod aperiam illum eos ad hic dolorum non nam ' +
            'voluptatem eveniet alias architecto fuga corrupti. Quos.';
        this.description = '';
        this.date = new Date();
    }
}